iroh/src/magicsock/transports/relay.rs

```diff
@@ -313,17 +315,16 @@ impl RelaySender {
 /// size, the contents will be sent as a single packet.
 // TODO: If quinn stayed on bytes this would probably be much cheaper, probably.  Need to
 // figure out where they allocate the Vec.
-fn split_packets(transmit: &Transmit<'_>) -> RelayContents {
-    let mut res = SmallVec::with_capacity(1);
-    let contents = transmit.contents;
-    if let Some(segment_size) = transmit.segment_size {
-        for chunk in contents.chunks(segment_size) {
-            res.push(Bytes::from(chunk.to_vec()));
-        }
-    } else {
-        res.push(Bytes::from(contents.to_vec()));
+fn datagrams_from_transmit(transmit: &Transmit<'_>) -> Datagrams {
+    Datagrams {
+        ecn: transmit.ecn.map(|ecn| match ecn {
+            quinn_udp::EcnCodepoint::Ect0 => quinn_proto::EcnCodepoint::Ect0,
+            quinn_udp::EcnCodepoint::Ect1 => quinn_proto::EcnCodepoint::Ect1,
+            quinn_udp::EcnCodepoint::Ce => quinn_proto::EcnCodepoint::Ce,
+        }),
+        segment_size: transmit.segment_size.map(|ss| ss as u16),
+        contents: Bytes::copy_from_slice(transmit.contents),
```

Comment:
We still have to copy?  :sob:
Why?

Reply:
Because `transmit.contents` has type `&[u8]`, i.e. we don't own the buffer contents, so we can't `try_send` them over a task boundary.